### ISSUE TITLE
Auto-generate transaction IDs when creating transactions

### ISSUE THREAD

**Issue by mjtravers** [fecgov/fecfile-web-api#124]
### Business Reason ### 

As a user creating a new transaction, I want a unique transaction id created when saving.

### Acceptance Criteria ### 

**If** a user saves a transaction
**When** it is saved
**Then** it will be assigned a unique transaction id. 

**And when** I am on the View All Transactions screen
**Then** I can see that the child is associated to a parent transaction using the following columns:
1. Transaction ID (**update to show actual Transaction ID**)
2. Associated with _(**update to show actual Transaction ID of parent**)_

### Tech Notes ###
The Transaction ID will be A/N-20 character limit.
The Transaction ID _must_ be unique within a committee account.
We will generate a UUID then truncate it to 20 characters, and before saving and storing we will ensure that it is not a duplicate of an existing transaction ID. If it is, another shall be generated and the same process followed.
The "repeat" process will max out and throw an error after 5 tries in the unlikely chance it finds duplicate transaction ids 5 times in a row.

### Dev Update ###
On further review with Matt, the UUID will be generated by the API using Python's [UUID.int](https://docs.python.org/3/library/uuid.html#:~:text=The%20UUID%20as%20a%2032%2Dcharacter%20lowercase%20hexadecimal%20string.&text=The%20UUID%20as%20a%20128%2Dbit%20integer.).  This guarantees uniqueness across the database while also being able to be represented by an 8-character hexadecimal value.

- See rules on COL SEQ 4 in the [OFFSET_TO_OPEX spec](https://fecgov.github.io/fecfile-validate/OFFSET_TO_OPEX_spec.html)
- New ID needs to be generated before validation takes place.
- This code can be removed from the front end when this feature is running in the back end: https://github.com/fecgov/fecfile-web-app/blob/feature/230-group-B-transactions/front-end/src/app/transactions/transaction-group-b/transaction-group-b.component.ts#L121-L124

### QA Notes ###
To test, every time a new transaction is created it will be assigned a transaction ID that must be unique across all transactions within the Committee Account. To verify the transaction ID, it will display in the list of transactions table for the report. This transaction ID will not match any other transaction ID for the Committee Account and it will stay the same even if the transaction is edited and updated.

**Comment by MitchellTCG** [comment]
@AureliaKhorsand  
Waiting on more info from 5/19 requirements meeting

**Comment by AureliaKhorsand** [comment]
**Refinement Meeting 6/1/22:** Shelly and Todd not present.

**Comment by AureliaKhorsand** [comment]
BA Note: Moving ahead with this solution, will maintain chronological order using date/time stamps in the system.

**Comment by mjtravers** [comment]
Passes CR. Sending to QA.
